kurtosis-module/impl/module_io/params.go

```diff
@@ -145,40 +155,42 @@ type ParticipantParams struct {
 	// Extra parameters that will be passed to the validator container (if a separate one exists), or to the combined node if
 	// the Beacon and validator are combined
 	ValidatorExtraParams []string `yaml:"validatorExtraParams"`
+
+	BuilderNetworkParams *BuilderNetworkParams `yaml:"builderNetworkParams"`
```

Comment:
Since we're adding a new param, you'll need to:
1) update the README with this new param, with default values added
2) update the param deserialization so that when the `builderNetworkParams` are specified, all the required variables are guaranteed to exist (e.g. in this case, I think we'd need to ensure that `if builderNetworkParams != nil` then we check that `Relays` is non-empty and the relays aren't emptystring)

Reply:
I would like to keep the option to have more stuff and I think "nesting" conceptually firewalls off this data from the rest of the struct which imo helps readers understand what is going on

I do think for now this is the only builder-wide configuration we will have and so I'm happy to inline if you prefer

Reply:
> update the param deserialization so that when the builderNetworkParams are specified, all the required variables are guaranteed to exist (e.g. in this case, I think we'd need to ensure that if builderNetworkParams != nil then we check that Relays is non-empty and the relays aren't emptystring)

does the yaml serialization library not handle this for us?

the consumers of this config (which are also added in this PR) already check if the struct is nil or not and use this to signal if they should proceed further w/ instantiating the service(s).

Reply:
> update the README with this new param, with default values added

if we use "nil as enabled/disable" then the default will be disabled and the readme does not need to be updated

Reply:
ah got it, so `nil` == "there is no network builder"; that makes sense to me